src/stirling/source_connectors/socket_tracer/data_stream.h

```diff
@@ -254,22 +270,42 @@ class DataStream : NotCopyMoveable {
   protocols::DataStreamBuffer& data_buffer() { return data_buffer_; }
 
  private:
-  template <typename TFrameType>
+  template <typename TKey, typename TFrameType>
   static void EraseExpiredFrames(
       std::chrono::time_point<std::chrono::steady_clock> expiry_timestamp,
-      std::deque<TFrameType>* frames) {
-    auto iter = frames->begin();
-    for (; iter != frames->end(); ++iter) {
-      auto frame_timestamp = std::chrono::time_point<std::chrono::steady_clock>(
-          std::chrono::nanoseconds(iter->timestamp_ns));
-      // As messages are put into the list with monotonically increasing creation time stamp,
-      // we can just stop at the first frame that is younger than the expiration duration.
-      // TODO(yzhao): Benchmark with binary search and pick the faster one.
-      if (expiry_timestamp < frame_timestamp) {
-        break;
+      std::map<TKey, std::deque<TFrameType>>* frames) {
+    LOG(INFO) << "Expiry timestamp: "
+              << std::chrono::duration_cast<std::chrono::milliseconds>(
+                     expiry_timestamp.time_since_epoch())
+                     .count()
+              << "ms";
+    for (auto it = frames->begin(); it != frames->end(); it++) {
+      auto stream = it->first;
+      LOG(INFO) << "Stream " << stream;
+      auto& deque = it->second;
+      LOG(INFO) << "Size " << deque.size();
+      auto iter = deque.begin();
+      for (; iter != deque.end(); ++iter) {
+        auto frame_timestamp = std::chrono::time_point<std::chrono::steady_clock>(
+            std::chrono::nanoseconds(iter->timestamp_ns));
+        LOG(INFO) << "Frame timestamp: "
+                  << std::chrono::duration_cast<std::chrono::milliseconds>(
+                         frame_timestamp.time_since_epoch())
+                         .count()
+                  << "ms";
+        // As messages are put into the list with monotonically increasing creation time stamp,
+        // we can just stop at the first frame that is younger than the expiration duration.
+        // TODO(yzhao): Benchmark with binary search and pick the faster one.
+        if (expiry_timestamp < frame_timestamp) {
+          LOG(INFO) << "Expired";
+          break;
+        }
       }
+
+      LOG(INFO) << "Distance " << std::distance(deque.begin(), iter);
+      deque.erase(deque.begin(), iter);
+      LOG(INFO) << "Size " << deque.size();
```

Comment:
These logs statements appear to be debugging remnants.

Reply:
Removed in most recent commit